python/lsst/afw/table/baseColumnView/baseColumnView.cc

```diff
@@ -0,0 +1,132 @@
+/*
+ * LSST Data Management System
+ * Copyright 2008-2017  AURA/LSST.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <https://www.lsstcorp.org/LegalNotices/>.
+ */
+
+#include "pybind11/pybind11.h"
+#include "pybind11/stl.h"
+
+#include "numpy/arrayobject.h"
+#include "ndarray/pybind11.h"
+
+#include "lsst/afw/table/Key.h"
+#include "lsst/afw/table/BaseColumnView.h"
+
+namespace ndarray { namespace detail {
+
+template <> struct NumpyTraits<lsst::afw::geom::Angle> : public NumpyTraits<double> {};
+
+}} // namespace ndarray::detail
+
+namespace py = pybind11;
+using namespace py::literals;
+
+namespace lsst {
+namespace afw {
+namespace table {
+namespace {
+
+using PyBaseColumnView = py::class_<BaseColumnView, std::shared_ptr<BaseColumnView>>;
+
+using PyBitsColumn = py::class_<BitsColumn, std::shared_ptr<BitsColumn>>;
+
+template <typename T, typename PyClass>
+static void declareBaseColumnViewOverloads(PyClass & cls) {
+    cls.def("_basicget", [](BaseColumnView & self, Key<T> const & key)
+                          ->typename ndarray::Array<T,1> const {
+        return self[key];
+    });
+};
+
+template <typename U, typename PyClass>
+static void declareBaseColumnViewArrayOverloads(PyClass & cls) {
+    cls.def("_basicget", [](BaseColumnView & self, Key<lsst::afw::table::Array<U>> const & key)
+                          ->typename ndarray::Array<U,2,1> const {
+        return self[key];
+    });
+};
+
+template <typename PyClass>
+static void declareBaseColumnViewFlagOverloads(PyClass & cls) {
+    cls.def("_basicget", [](BaseColumnView & self, Key<Flag> const & key)
+                          ->ndarray::Array<bool const,1,1> const {
+        return ndarray::copy(self[key]);
+    });
+};
+
+static void declareBaseColumnView(py::module & mod) {
+    PyBaseColumnView cls(mod, "_BaseColumnViewBase"); // "BaseColumnView" = "ColumnViewT<BaseRecord>"
```

Comment:
`BaseColumnView` is more logically mapped to `ColumnViewT<BaseRecord>` in Python, so I'm using the same naming convention I use for invisible base classes in the `declare` methods.  Comment expanded.